contrib/IECoreUSD/src/IECoreUSD/USDScene.cpp

```diff
@@ -190,13 +190,56 @@ IECoreScene::PointsPrimitivePtr convertPrimitive( pxr::UsdGeomPointInstancer poi
 	return newPoints;
 }
 
-IECoreScene::CurvesPrimitivePtr convertPrimitive( pxr::UsdGeomCurves curves, pxr::UsdTimeCode time )
+IECoreScene::CurvesPrimitivePtr convertPrimitive( pxr::UsdGeomBasisCurves curves, pxr::UsdTimeCode time )
 {
 	pxr::VtIntArray vertexCountsArray;
 	curves.GetCurveVertexCountsAttr().Get( &vertexCountsArray, time );
 	IECore::IntVectorDataPtr countData = DataAlgo::fromUSD( vertexCountsArray );
 
-	IECoreScene::CurvesPrimitivePtr newCurves = new IECoreScene::CurvesPrimitive( countData, IECore::CubicBasisf::linear(), false );
+	// Basis
+
+	IECore::CubicBasisf basis = CubicBasisf::linear();
+	pxr::TfToken type;
+	curves.GetTypeAttr().Get( &type, time );
+	if( type == pxr::UsdGeomTokens->cubic )
+	{
+		pxr::TfToken usdBasis;
+		curves.GetBasisAttr().Get( &usdBasis, time );
+		if( usdBasis == pxr::UsdGeomTokens->bezier )
+		{
+			basis = CubicBasisf::bezier();
+		}
+		else if( usdBasis == pxr::UsdGeomTokens->bspline )
+		{
+			basis = CubicBasisf::bSpline();
+		}
+		else if( usdBasis == pxr::UsdGeomTokens->catmullRom )
+		{
+			basis = CubicBasisf::catmullRom();
+		}
+		else
+		{
+			IECore::msg( IECore::Msg::Warning, "USDScene", boost::format( "Unsupported basis \"%1%\"" ) % usdBasis );
+		}
+	}
+
+	// Wrap
+
+	bool periodic = false;
+	pxr::TfToken wrap;
+	curves.GetWrapAttr().Get( &wrap, time );
+	if( wrap == pxr::UsdGeomTokens->periodic )
+	{
+		periodic = true;
+	}
+	else if( wrap != pxr::UsdGeomTokens->nonperiodic )
```

Comment:
That's it, but to support it well we wouldn't want to call `updateEndpointMultiplicity()` on read, we'd want to be able to pass the "pinned" flag around through Cortex and only add the phantom points when hitting the renderer. We have a similar thing going on with `Gaffer::SplineDefinition` vs `IECore::Spline` and we'd be better off to deal with it all holistically.

Reply:
To clarify, are you saying you're intending to take the holistic approach as part of your current "get USD working well in Gaffer" effort, or leave it as an idea for the future?

If the later, does this mean that, for the time being, pinned USD curves authored in Maya/Houdini will appear to have different endpoints when viewed in Gaffer's Viewer and when rendered to Arnold via Gaffer?

Reply:
I haven't got a strong position on that yet, and I'm still waiting for example data from production for Maya/Houdini exports to see if/how the pinning is taken advantage of. My preference would be to take the more holistic approach, but like many things, it will need to wait for #1058. My current plan is to move on to other aspects of the USD/Alembic support and circle back when I know more.